web-client/src/presenter/sequences/signOutIdleSequence.ts

```diff
@@ -0,0 +1,20 @@
+import { LOGOUT_BROADCAST_MESSAGES } from '@shared/business/entities/EntityConstants';
+import { isLoggedInAction } from '@web-client/presenter/actions/isLoggedInAction';
+import { setLogoutTypeAction } from '@web-client/presenter/actions/setLogoutTypeAction';
+import { setupCurrentPageAction } from '../actions/setupCurrentPageAction';
+import { signOutSequence } from '@web-client/presenter/sequences/signOutSequence';
+
+// The sequence to call when the user is forced to sign out due to idle activity
+export const signOutIdleSequence = [
+  isLoggedInAction,
+  {
+    // To avoid race conditions, we ignore redundant calls to sign out that arise when
+    // multiple tabs broadcast the idle sign out event.
+    no: [],
+    yes: [
+      setLogoutTypeAction(LOGOUT_BROADCAST_MESSAGES.idleLogout),
+      signOutSequence,
+      setupCurrentPageAction('IdleLogout'),
+    ],
```

Comment:
The issue: all tabs/windows will time out at the same time and send a broadcast message to the others to log out as well, which can lead to race issues (e.g., an infinite loading screen).

If someone has a better solution, I'm all ears! The other solutions I considered are either insufficient or more involved:

1) We use `window.document.visibilityState` to only broadcast from visible tabs. However, this doesn't solve the issue since the user might have multiple visible windows. Moreover, I am afraid of something like the following occurring: user starts to log out but something hangs; tab is minimized in the meantime; no broadcast happens, and therefore sign out fails in other tabs.
2) Do something similar, but set some local state to keep track of the last focused tab and only broadcast from it. This might be a good solution but requires more work and seems more prone to error (now we have to keep track of tabs--e.g., what happens when the last active tab is closed?).

Redundant calls to sign out aren't ideal, but they are already happening _without_ being accounted for.